native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -1,20 +1,64 @@
 import UIKit
 import SharedCode
 
-class ViewController: UIViewController {
 
+class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource, ApplicationContractView {
+    
     @IBOutlet private var label: UILabel!
-
-    private let presenter: ApplicationContractPresenter = ApplicationPresenter()
+    @IBOutlet weak var departurePicker: UIPickerView!
+    @IBOutlet weak var arrivalPicker: UIPickerView!
     
+    private let presenter: ApplicationContractPresenter = ApplicationPresenter()
+    var stationData: [String] = [String]()
     override func viewDidLoad() {
         super.viewDidLoad()
         presenter.onViewTaken(view: self)
+        
     }
-}
-
-extension ViewController: ApplicationContractView {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return stationData.count
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return stationData[row]
+    }
+    
+    
+    @IBAction func departureButton(_ sender: Any) {
+        let departureStation = stationData[departurePicker.selectedRow(inComponent: 0)]
+        let arrivalStation = stationData[arrivalPicker.selectedRow(inComponent: 0)]
+        let urlString: String = "https://www.lner.co.uk/travel-information/travelling-now/live-train-times/depart/\(departureStation)/\(arrivalStation)/#LiveDepResults"
```

Comment:
As I said on the android side, I would expect this logic to live in the presenter rather than in the views 🙂